csrc/scheduler/greedy.cpp

```diff
@@ -72,16 +81,55 @@ std::unique_ptr<HeuristicParams> GreedyParams::clone() const {
   return std::make_unique<GreedyParams>(*this);
 }
 
-void GreedyParams::transferParams(TensorView* old_tv, TensorView* new_tv) {
-  if (auto it = tv_to_item_per_thread.find(old_tv->name());
-      it != tv_to_item_per_thread.end()) {
-    NVF_ERROR(
-        tv_to_item_per_thread.emplace(new_tv->name(), it->second).second,
-        "Duplicated setting for ",
-        new_tv->toString());
-    // Remove the old entry
-    tv_to_item_per_thread.erase(old_tv->name());
+bool GreedyParams::hasConsumerParams(TensorView* consumer_tv) {
+  return consumer_to_params_.contains(consumer_tv->name());
+}
+
+bool GreedyParams::hasProducerParams(
+    TensorView* producer_tv,
+    TensorView* consumer_tv) const {
+  return producer_to_params_.contains(
+      std::make_pair(producer_tv->name(), consumer_tv->name()));
+}
+
+void GreedyParams::transferConsumerParams(
+    TensorView* old_tv,
+    TensorView* new_tv) {
+  auto it = consumer_to_params_.find(old_tv->name());
+  if (it == consumer_to_params_.end()) {
+    return;
   }
+  NVF_ERROR(
+      setConsumerParams(new_tv, it->second),
+      "Duplicated setting for ",
+      new_tv->toString());
+  // Remove the old entry
+  consumer_to_params_.erase(old_tv->name());
+}
+
+void GreedyParams::transferProducerParams(
+    TensorView* old_producer_tv,
+    TensorView* old_consumer_tv,
+    TensorView* new_producer_tv,
+    TensorView* new_consumer_tv) {
+  NVF_ERROR(old_producer_tv != nullptr);
+  NVF_ERROR(old_consumer_tv != nullptr);
+  NVF_ERROR(new_producer_tv != nullptr);
+  NVF_ERROR(new_consumer_tv != nullptr);
```

Comment:
nitpick: among all the transfer / query, transferProducerParams is the only one that asserts on input not being nullptr. which felt a bit strange not having this on the others.